target.c

```diff
@@ -329,8 +330,13 @@ int tcmu_add_dev_to_recovery_list(struct tcmu_device *dev)
 		ret = -ENOMEM;
 		goto done;
 	}
-	ret = pthread_create(&tpg->recovery_thread, NULL,
+
+	pthread_attr_init(&attr);
+	pthread_attr_setdetachstate(&attr, PTHREAD_CREATE_DETACHED);
+
+	ret = pthread_create(&thread_id, &attr,
```

Comment:
If this tread is just running in the background, how is it tracked for joining before the process exits? Why not re-use the new `tcmur_work` thread wrapper and add the work struct to `tgt_port_grp`?

Reply:
Yeah, good catch. In linux the detached thread will be terminated when the main process is exiting.  But there still have one problem, if the main process is exiting and it will release all the resources before it terminates all the detached threads, which may still accessing them.

I will convert to the tcmur_work thread wrapper, but the tcmur_work thread wrapper has one bug, I will fix it at the same time.